workflow/controller/workflowpod.go

```diff
@@ -764,22 +764,33 @@ func (woc *wfOperationCtx) addMetadata(pod *apiv1.Pod, tmpl *wfv1.Template) {
 }
 
 // addSchedulingConstraints applies any node selectors or affinity rules to the pod, either set in the workflow or the template
-func addSchedulingConstraints(pod *apiv1.Pod, wfSpec *wfv1.WorkflowSpec, tmpl *wfv1.Template) {
+func (woc *wfOperationCtx) addSchedulingConstraints(pod *apiv1.Pod, wfSpec *wfv1.WorkflowSpec, tmpl *wfv1.Template, nodeName string) {
```

Comment:
Potentially switching to nodeID instead of nodeName makes some of this slightly simpler

Reply:
OK, thanks I will.

Reply:
I find it better to use nodeName to find the Boundary because we have GetNodeByName.

Reply:
Ok, what you've done with two functions below makes sense. I like it.